tests/float12.dora

```diff
@@ -4,4 +4,7 @@ fun main() {
 
     let x = 2.5D;
     assert(2.5F == x.toFloat());
+
+    let x = 3;
+    assert(4.2E-45F == x.asFloat());
 }
```

Comment:
We should add here a few more simple test cases, like `0`, `-0` or `NaN`. Those are values with well known bit representations. In both directions (float to int and int to float) and also for float and double.